Data/Text/Internal/Fusion/Common.hs

```diff
@@ -699,7 +702,7 @@ replicateI n (Stream next0 s0 len) =
 -- (a,b), in which case, a is the next Char in the string, and b is
 -- the seed value for further production.
 unfoldr :: (a -> Maybe (Char,a)) -> a -> Stream Char
-unfoldr f s0 = Stream next s0 1 -- HINT maybe too low
+unfoldr f s0 = Stream next s0 unknownSize
```

Comment:
Don't you drop `s` in ever stream, so can't `unknownSize` be `charSize s`?

Reply:
I'm not sure I follow. Can you rephrase?

Reply:
What I mean is, in each step of the stream you consume one input character don't you? So according to the comment

     -- The function takes the element and returns
     -- Nothing if it is done producing the stream or returns Just
     -- (a,b), in which case, a is the next Char in the string, and b is 
    -- the seed value for further production. 

So isn't the initial length the number of code points of the seed?

Reply:
Ahh, I see.
> So isn't the initial length the number of code points of the seed?

Not quite, since the user might take and double the size of each code point (turning one-unit code-point into a two-unit code point). Consequently, I think we can set the upper bound at `2*hint`.

Reply:
Let's try this again.

> What I mean is, in each step of the stream you consume one input character don't you?

This is where you lose me. What input stream are we talking about here? `unfoldr` produces a stream by unfolding starting from an initial value, successively applying the given function until it returns `Nothing`. Consequently I'm not sure how we could possibly know how long the stream will be; it depends entirely on the function.

Reply:
Ah, there's my misunderstanding. I had thought it only applied it once to get the next value. In which case I figured you always know the max amount of code points the next would produce. Thanks for clearing that up!